### PR TITLE
Document IAM policy for s3, move s3 info to dedicated page

### PR DESCRIPTION
**Pull request by mdisibio**
**What this PR does**:
Document minimal IAM permissions for s3 usage.  Create dedicated page for all s3 configuration info.

**Which issue(s) this PR fixes**:
n/a

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by joe-elliott — APPROVED** [review]

File: docs/tempo/website/configuration/_index.md [hunk 1]
```diff
@@ -68,24 +68,19 @@ compactor:
 ```
 
 ## [Storage](https://github.com/grafana/tempo/blob/master/tempodb/config.go)
-The storage block is used to configure TempoDB.
+The storage block is used to configure TempoDB.  It supports S3, GCS, local file system, and optionally can use memcached for increased query performance.
```
Comment (achatterjee-grafana) [inline docs/tempo/website/configuration/_index.md:71]:
```suggestion
The storage block is used to configure TempoDB. It supports S3, GCS, local file system, and optionally can use memcached for increased query performance.  
```

File: docs/tempo/website/configuration/_index.md [hunk 2]
```diff
@@ -68,24 +68,19 @@ compactor:
...
 
-For the s3 backend, the following authentication methods are supported:
-
-- AWS env vars (static AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY)
-- Static access key and secret credentials specified in `storage.trace.s3.access_key` and `storage.trace.s3.secret_key`
-- Minio env vars (MINIO_ACCESS_KEY and MINIO_SECRET_KEY)
-- AWS shared credentials [config file](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/create-shared-credentials-file.html)
-- Minio client credentials [config file](https://github.com/minio/mc/blob/master/docs/minio-client-configuration-files.md)
-- AWS IAM ([IRSA via WebIdentity](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html), [EC2 instance role](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html))
+The example below shows common options.  For platform specific options see the following:
```
Comment (achatterjee-grafana) [inline docs/tempo/website/configuration/_index.md:73]:
```suggestion
The following example shows common options.  For platform-specific options refer to the following:
```

File: docs/tempo/website/configuration/s3.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                "arn:aws:s3:::<bucketname>/*",
+                "arn:aws:s3:::<bucketname>"
+            ]
+        }
+    ]
+}
+```
+
+### Lifecycle Policy
+A lifecycle policy is recommended that deletes incomplete multipart uploads after 1 day.
```
Comment (achatterjee-grafana) [inline docs/tempo/website/configuration/s3.md:55]:
```suggestion
A lifecycle policy is recommended that deletes incomplete multipart uploads after one day.
```

File: docs/tempo/website/configuration/s3.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            secret_key: ...                                 # optional. secret key when using static credentials.
+            insecure: false                                 # optional. enable if endpoint is http
+```
+
+### Permissions
+The following authentication methods are supported:
+- AWS environment variables AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY
+- Static access key and secret credentials specified in `access_key` and `secret_key`
+- MinIO environment variables MINIO_ACCESS_KEY and MINIO_SECRET_KEY
+- AWS shared credentials [config file](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/create-shared-credentials-file.html)
```
Comment (achatterjee-grafana) [inline docs/tempo/website/configuration/s3.md:26]:
```suggestion
- AWS shared credentials [configuration file](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/create-shared-credentials-file.html)
```

File: docs/tempo/website/configuration/s3.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+- MinIO client credentials [config file](https://github.com/minio/mc/blob/master/docs/minio-client-configuration-files.md)
```
Comment (achatterjee-grafana) [inline docs/tempo/website/configuration/s3.md:27]:
```suggestion
- MinIO client credentials [configuration file](https://github.com/minio/mc/blob/master/docs/minio-client-configuration-files.md)
```

File: docs/tempo/website/configuration/s3.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+- AWS IAM ([IRSA via WebIdentity](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html), [EC2 instance role](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html))
```
Comment (achatterjee-grafana) [inline docs/tempo/website/configuration/s3.md:28]:
```suggestion
- AWS IAM ([IRSA via WebIdentity](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html), 
- AWS [EC2 instance role](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html))
```

File: docs/tempo/website/configuration/s3.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
+---
+title: S3
+---
+
+### S3 Configuration
```
Comment (achatterjee-grafana) [inline docs/tempo/website/configuration/s3.md:5]:
```suggestion
# S3 Configuration
```

File: docs/tempo/website/configuration/s3.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+        s3:
+            bucket: tempo                                   # store traces in this bucket
+            endpoint: s3.dualstack.us-east-2.amazonaws.com  # api endpoint
+            region: us-east-2                               # optional. region in which to create bucket, if needed
+            access_key: ...                                 # optional. access key when using static credentials. 
+            secret_key: ...                                 # optional. secret key when using static credentials.
+            insecure: false                                 # optional. enable if endpoint is http
+```
+
+### Permissions
```
Comment (achatterjee-grafana) [inline docs/tempo/website/configuration/s3.md:21]:
```suggestion
## Permissions
```

File: docs/tempo/website/configuration/s3.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+            "Resource": [
+                "arn:aws:s3:::<bucketname>/*",
+                "arn:aws:s3:::<bucketname>"
+            ]
+        }
+    ]
+}
+```
+
+### Lifecycle Policy
```
Comment (achatterjee-grafana) [inline docs/tempo/website/configuration/s3.md:54]:
```suggestion
## Lifecycle Policy
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added some copy-edit suggestions.

**Review by achatterjee-grafana — APPROVED** [review]
Approving, but please add the copy-edit changes. Thanks!

**Pull request merged**